proto/redpanda/api/dataplane/v1alpha2/cloud_storage.proto

```diff
@@ -0,0 +1,393 @@
+    // Cut Over: The mount task has reached a critical point where ownership is transferred or final adjustments are made.
+    STATE_CUT_OVER = 6;
+    // Finished: The mount task has been successfully completed, and no further actions are required.
+    STATE_FINISHED = 7;
+    // Canceling: The mount task is in the process of being canceled, and rollback or cleanup actions may be in progress.
+    STATE_CANCELING = 8;
+    // Cancelled: The mount task has been fully canceled, and no further actions will be taken.
+    STATE_CANCELLED = 9;
+  }
+
+  // Id is the unique identifier for this mount task.
+  int32 id = 1;
+  // State describes the current state of the mount task (e.g. "cancelled").
+  State state = 2;
+  // Type describes the type of this task (mount or unmount).
+  Type type = 3;
+  // Topics is the list of topics that are being mounted or unmounted.
+  repeated Topic topics = 4;
+}
+
+message ListMountTasksRequest {}
+
+message ListMountTasksResponse {
+  repeated MountTask tasks = 1;
+}
+
+message GetMountTaskRequest {
+  // ID is the unique identifier of the mount task to retrieve.
+  int32 id = 1 [
+    // We won't add validation required here, because 0 may be a valid id.
+    (google.api.field_behavior) = REQUIRED
+  ];
+}
+
+message GetMountTaskResponse {
+  MountTask task = 1;
+}
+
+message DeleteMountTaskRequest {
+  // ID is the unique identifier of the mount task to delete.
+  int32 id = 1 [(google.api.field_behavior) = REQUIRED /* We won't add validation required here, because 0 may be a valid id. */];
+}
+
+message DeleteMountTaskResponse {}
+
+message UpdateMountTaskRequest {
+  enum Action {
+    ACTION_UNSPECIFIED = 0;
+    ACTION_PREPARE = 1;
+    ACTION_EXECUTE = 2;
+    ACTION_FINISH = 3;
+    ACTION_CANCEL = 4;
+  }
+
+  // ID is the unique identifier of the mount task to delete.
+  int32 id = 1 [(google.api.field_behavior) = REQUIRED /* We won't add validation required here, because 0 may be a valid id. */];
+
+  // Action is the action to execute on mount task.
+  Action action = 2 [
+    (google.api.field_behavior) = REQUIRED,
+    (buf.validate.field).required = true,
+    (buf.validate.field).enum = {
+      defined_only: true
+      not_in: [0] /* Disallow: Unspecified */
+    }
+  ];
+}
+
+message UpdateMountTaskResponse {}
+
+service CloudStorageService {
+  rpc MountTopics(MountTopicsRequest) returns (MountTopicsResponse) {
+    option (google.api.http) = {
+      post: "/v1alpha2/cloud-storage/topics/mount"
+      body: "topics"
+      response_body: "*"
+    };
+    option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_operation) = {
+      summary: "Mount topics from tiered storage"
+      description: "This operation restores topics that were offloaded to tiered storage, making them available for consumption and production again. Remounting a topic reloads its data and state to the local brokers, allowing active use of the topic while consuming some cluster resources."
```

Comment:
```suggestion
      description: "Attaches mountable topics from object storage to a cluster, making them available for consumption and production again. Mounting a topic reloads its data and state to the local brokers, allowing active use of the topic."
```
I hesitate to say "restore" because that could be taken to mean in the DR sense, what do you think? I'm also curious to know what "consuming some cluster resources" means exactly, and if it might be OK to just leave the phrase out entirely.

Reply:
> consuming some cluster resources

Each partition consumes some RAM / CPU / Disk resources. Hard to quantify this, but people should generally consider this before mounting, that's why I mentioned it here. Managed clusters also have enforced limits such as partition count etc.